### ISSUE TITLE
prefect 2.0 deployment build does not get current directory

### ISSUE THREAD

**Issue by wangjoshuah** [PrefectHQ/prefect#6111]
I'm trying to build a prefect deployment in 2.0 and it will not recognize my local path when building so I can't import files in subdirectories.

Setup, I'm using python 3.9 and poetry to manage my [project](https://github.com/wangjoshuah/prefect_certification/tree/4b0dc387cf85b8cb1d9618821232f5ea71c49967)

after cloning, just try 

```bash
prefect deployment build ./src/flows/ingest_pokemon_data.py:ingest_pokemon_data -n pokemon -t test
```

Stacktrace:

```
File "./src/flows/ingest_pokemon_data.py", line 10, in <module>
    from src.blocks.pokemon_api_block import PokemonApiBlock, pokemon_api_block_key
ModuleNotFoundError: No module named 'src'
```

Workaround:
```bash
PYTHONPATH=(pwd) prefect deployment build ./src/flows/ingest_pokemon_data.py:ingest_pokemon_data -n pokemon -t test
```

**Comment by johnkangw** [comment]
@wangjoshuah Thanks for raising the issue! Curious what the resolution will be.

**Comment by barino86** [comment]
Does Prefect allow users to keep their deployment files (*-deployment.yaml & *-manifest.json) in a separate directory other than the actual root directory of the project to keep things organized in a project with multiple flows? I am specifically asking about the S3 storage block.

Example project structure:
```
root
|-- deployments
|    |-- flow1-deployment.yaml
|    |-- flow1-manifest.json
|    |-- flow2-deployment.yaml
|    |-- flow2-manifest.json
|-- src
|    |-- flow1
|    |    |-- flow1.py
|    |-- flow2
|    |    |-- flow2.py
|-- README.md
|-- Other-file.txt
```

From the source code, it looks like the build command only copies from the current directory all recursive files. So if you run the build command in the `deployments` folder it won't copy any actual flow files.

**Comment by wangjoshuah** [comment]
> Does Prefect allow users to keep their deployment files (*-deployment.yaml & *-manifest.json) in a separate directory other than the actual root directory of the project to keep things organized in a project with multiple flows?

I think this is a great feature request. Something like `prefect deployment build -o` or `prefect deployment build --output_path`. You can try to submit it as a feature request in the [Discourse](https://discourse.prefect.io/)

**Comment by drfraser** [comment]
I have just run into this issue myself while trying to set up my flows/deployments.  I tried using "prefect deployment build ...  --output deployments/test.yaml" and the yaml file went into ./deployments while the manifest file was placed into the root directory which was unexpected.  

The deployment then ran successfully which told me the working directory for the deployment was the root (my flow reads a config file and it was found successfully unlike other attempts). So I'd be happy if the manifest file was placed where the deployment yaml was.

I did try editing import_path and basepath/manifest_path to try and arrange things as I wanted, but that doesn't seem to work...  Perhaps I missed it, but an explanation in the docs of how deployments work internally would be useful
